data/docs/how-to-add-test-case.md

```diff
@@ -0,0 +1,146 @@
+## 1. Create an environment for testing
+
+When using Rector to update your own code, you will typically be using release repository installed by composer, however, when adding tests, you will need to use the development repository as shown:
+
+- Fork https://github.com/rectorphp/rector-src
+- Clone it locally
+- Install dependencies by executing `composer install`
+- Tests your installation by executing `composer fix-cs` and `composer phpstan`
+- Create a new branch for your test
+- Add your test as described below. Note that the rector binary is located at `bin/rector` instead of the typical `vendor/bin/rector`
+- Push the branch
+- Create a new pull request to https://github.com/rectorphp/rector-src
+
+<br>
+
+## 2. Detect the Rector Rule
+
+Run Rector only on 1 directory, or better 1 file.
+
+```bash
+bin/rector process /some-file.php
+```
+
+See "Applied rules" under the diff:
+
+![Applied Rules](/docs/images/docs_applied_rules.png)
+
+Our rule in this example is: `Rector\Privatization\Rector\Class_\FinalizeClassesWithoutChildrenRector`
+
+This rule's job is to add `final` to every class that has no children and is not a Doctrine entity = everywhere it can without breaking our code.
+
+<br>
+
+## 3. Detect the Minimal File
+
+Usually, the Rector diff output is long and contains many other errors related to other rules. It's a mess; we can't use that for a test fixture. We need to find **1 responsible line**.
+
+The best way is to copy the file to local code, e.g. `app/SomeFile.php` and put only the broken line there.
+
+In our case, all we need is:
+
+```php
+class StaticEasyPrefixer
+{
+}
+```
+
+Then rerun Rector to confirm:
+
+```bash
+bin/rector process app/SomeFile.php
+```
+
+Do we have the same diff? Great!
+
+<br>
+
+## 4. Find the Rector Test Case
+
+Now we need to find the test case. The test case name is rule + `Test` suffix.
+
+`FinalizeClassesWithoutChildrenRector`
+
+↓
+
+`FinalizeClassesWithoutChildrenRectorTest` (test class)
+
+Right here:
+
+![Rule Test Case](/docs/images/docs_rule_test_case.png)
```

Comment:
images seems need to be imported, or use raw uri?  https://github.com/rectorphp/rector/blob/main/docs/images/docs_applied_rules.png

Reply:
I've missed to move it here, thanks :+1:

Reply:
See https://github.com/rectorphp/getrector.org/pull/904